test/app/exercises_test.rb

```diff
@@ -9,11 +9,31 @@ def app
     ExercismWeb::App
   end
 
-  def test_excercises_by_key
-    alice = User.create(username: 'alice', github_id: 1)
-    exercise = UserExercise.create(user: alice, language: 'go', slug: 'one')
-    submission = Submission.create(user: alice, language: 'go', slug: 'bob', user_exercise: exercise)
+  attr_reader :alice, :exercise, :submission
+
+  def setup
+    super
+    @alice = User.create(username: 'alice', github_id: 1)
+    @exercise = UserExercise.create(user: alice, language: 'go', slug: 'one',
+                                    last_activity_at: Date.today, iteration_count: 1)
+    @submission = Submission.create(user: alice, language: 'go',
+                                    slug: 'one', user_exercise: exercise)
+  end
+
+  def test_exercises_next_without_session_values
+    bob = User.create(username: 'bob', github_id: 1)
+    next_exercise = UserExercise.create(user: bob, language: 'go', slug: 'one',
+                                        last_activity_at: Date.yesterday,
+                                        iteration_count: 1)
+    ACL.create(user_id: alice.id, language: 'go', slug: 'one')
+
+    get "/exercises/next", {}, login(alice)
+    assert_equal 302, last_response.status
+    location = "http://example.org/exercises/#{next_exercise.key}"
+    assert_equal location, last_response.location, "Wrong redirect for GET exercises"
```

Comment:
Excellent--no hurry. Let me know if you get too swamped, I can take a stab at tweaking it and merging locally.